projects/ion/src/lib/picker/date-picker/date-picker-calendar/date-picker-calendar.component.ts

```diff
@@ -4,68 +4,67 @@ import {
   OnInit,
   Output,
   EventEmitter,
-  AfterViewInit,
+  DoCheck,
 } from '@angular/core';
-import { SafeAny } from '../../utils/safe-any';
-import { Calendar } from '../core/calendar';
-import { Day } from '../core/day';
+import { SafeAny } from '../../../utils/safe-any';
+import { Calendar } from '../../core/calendar';
+import { Day } from '../../core/day';
 
 type DateEmitter = {
-  date: string;
+  day: Day;
 };
 
-type DateField = {
-  date: Day;
-  label: string;
+export type UpdateLabelCalendar = {
+  month: string;
+  year: string;
 };
 
-export interface IonDatePickerProps {
-  isCalendarVisible?: boolean;
-  initialDate?: string;
+type CalendarControlActions =
```

Comment:
This is almost the same as TypeEvents type that you create it above. You could either use Pick<> or Omit<> to create this type or modify TypeEvents to extend this one.